package.json

```diff
@@ -13,11 +12,39 @@
     "bcrypt": "^5.0.1",
     "cookie-parser": "^1.4.5",
     "express": "^4.17.1",
+    "express-generator-typescript": "^1.7.8",
     "jsonwebtoken": "^8.5.1",
     "mongoose": "^5.12.11"
   },
+  "nodemonConfig": {
+    "watch": [
+      "src"
+    ],
+    "ext": "ts, html",
+    "ignore": [
+      "src/public"
+    ],
+    "exec": "./node_modules/.bin/ts-node -r tsconfig-paths/register ./src"
+  },
   "devDependencies": {
+    "@types/bcrypt": "^5.0.0",
```

Comment:
In project root folder you can add a `.npmrc` file with content `save-exact=true`. Then remove all `^` from versions. This way we are 100% sure we have the package version that is defined in package.json. Otherwise with each `npm i` call new minor version is updated. This has it's pros and cons.